cdap-watchdog/src/main/java/co/cask/cdap/logging/write/LogCleanup.java

```diff
@@ -52,28 +55,30 @@
 public final class LogCleanup implements Runnable {
   private static final Logger LOG = LoggerFactory.getLogger(LogCleanup.class);
   private static final int MAX_DISK_FILES_SCANNED = 50000;
-  private static final int MAX_META_FILES_SCANNED = 1000;
 
   private final FileMetaDataManager fileMetaDataManager;
   private final RootLocationFactory rootLocationFactory;
   private final String logBaseDir;
   private final NamespaceQueryAdmin namespaceQueryAdmin;
   private final NamespacedLocationFactory namespacedLocationFactory;
   private final long retentionDurationMs;
+  private final long maxMetaFileScanned;
   private final Impersonator impersonator;
 
   // this class takes a root location factory because for custom mapped namespaces the namespace is mapped to a
   // location from the root of system and the logs are generated in the custom mapped location. To clean up  these
   // locations we need to work with root based location factory
   public LogCleanup(FileMetaDataManager fileMetaDataManager, RootLocationFactory rootLocationFactory,
                     NamespaceQueryAdmin namespaceQueryAdmin, NamespacedLocationFactory namespacedLocationFactory,
-                    String logBaseDir, long retentionDurationMs, Impersonator impersonator) {
+                    String logBaseDir, long retentionDurationMs, CConfiguration cConf, Impersonator impersonator) {
     this.fileMetaDataManager = fileMetaDataManager;
     this.rootLocationFactory = rootLocationFactory;
     this.namespaceQueryAdmin = namespaceQueryAdmin;
     this.namespacedLocationFactory = namespacedLocationFactory;
     this.logBaseDir = logBaseDir;
     this.retentionDurationMs = retentionDurationMs;
+    this.maxMetaFileScanned = cConf.getLong(LoggingConfiguration.LOG_CLEANUP_MAX_NUM_FILES,
```

Comment:
per our convention, defaults should come from cdap-default.xml, not code. You added it there, so you don't need the default here.